From 158df4b6a63c0d4e4f8de5210581619e7b693fb4 Mon Sep 17 00:00:00 2001
From: Marius Thesing <marius.thesing@gmail.com>
Date: Tue, 8 Sep 2015 21:25:05 +0200
Subject: [PATCH] gpt-auto: fix ESP options and description

Since 59512f21 the parameters were passed in the wrong order, causing the
options to be interpreted as the description.

Also, while "false" was supposed to be passed for "rw", the ESP should actually
be mounted read-write. It just happened to be "true" since the description char*
was passed for "rw".
---
 src/gpt-auto-generator/gpt-auto-generator.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gpt-auto-generator/gpt-auto-generator.c b/src/gpt-auto-generator/gpt-auto-generator.c
index c97be2d..bb82179 100644
--- a/src/gpt-auto-generator/gpt-auto-generator.c
+++ b/src/gpt-auto-generator/gpt-auto-generator.c
@@ -526,9 +526,9 @@ static int add_boot(const char *what) {
                        what,
                        "/boot",
                        "vfat",
-                       "EFI System Partition Automount",
-                       false,
+                       true,
                        "umask=0077",
+                       "EFI System Partition Automount",
                        120 * USEC_PER_SEC);
 
         return r;
-- 
2.5.1

